src/LinearRemapFV.cpp

```diff
@@ -1611,199 +1610,451 @@ void LinearRemapGeneralizedBarycentric(
-						faceTriangleM.SetNode(2,2);
-
-						faceSubTriangleM.SetNode(0,0);
-						faceSubTriangleM.SetNode(1,1);
-						faceSubTriangleM.SetNode(2,2);
-
-						NodeVector nodesTriangleM;
-
-						NodeVector nodesSubTriangleM;
-
-						Node nodeM(meshInputDual.nodes[meshInputDual.faces[iFaceFinal][m]].x,
-								   meshInputDual.nodes[meshInputDual.faces[iFaceFinal][m]].y,
-								   meshInputDual.nodes[meshInputDual.faces[iFaceFinal][m]].z);
-
-						//c++ doesn't like the modulo operator when the argument is negative
-
-						int iMMinusOne = m - 1;
-
-						if (m == 0){
-
-							iMMinusOne = iEdges - 1;
-
-						}
-
+			//Get the dual mesh face whose center is nearest to the target face
 
-						Node nodeMMinusOne(meshInputDual.nodes[meshInputDual.faces[iFaceFinal][iMMinusOne]].x,
-										   meshInputDual.nodes[meshInputDual.faces[iFaceFinal][iMMinusOne]].y,
-										   meshInputDual.nodes[meshInputDual.faces[iFaceFinal][iMMinusOne]].z);
+			kdres * kdresTarget =
+				kd_nearest3(
+					kdDual,
+					nodeQ.x,
+					nodeQ.y,
+					nodeQ.z);
 
-						Node nodeMPlusOne(meshInputDual.nodes[meshInputDual.faces[iFaceFinal][(m+1)%iEdges]].x,
-										  meshInputDual.nodes[meshInputDual.faces[iFaceFinal][(m+1)%iEdges]].y,
-										  meshInputDual.nodes[meshInputDual.faces[iFaceFinal][(m+1)%iEdges]].z);
+			Face * pFace = (Face *)(kd_res_item_data(kdresTarget));
 
-						nodesTriangleM.push_back(nodeMMinusOne);
-						nodesTriangleM.push_back(nodeM);
-						nodesTriangleM.push_back(nodeMPlusOne);
+			int iNearestFace = pFace - &(meshInputDual.faces[0]);
 
-						nodesSubTriangleM.push_back(nodeQ);
-						nodesSubTriangleM.push_back(nodeM);
-						nodesSubTriangleM.push_back(nodeMPlusOne);
+			// Find triangle that contains nodeQ.
+			// This is the face whose local index is iFaceFinal
 
-						double dSubAreaM = CalculateFaceArea(faceSubTriangleM, nodesSubTriangleM);
+			int iFaceFinal = 0;
+
+			GetFaceThatContainsPoint(meshInputDual,iNearestFace,iFaceFinal,nodeQ.x,nodeQ.y,nodeQ.z);
+			
+			int iEdges = meshInputDual.faces[iFaceFinal].edges.size();
+			
+			//Current dual mesh face
+			
+			Face faceDualFace = meshInputDual.faces[iFaceFinal];
+			
+			//Vector of nodes on current dual face
+			
+			NodeVector nodesDualFace;
+			
+			for (int i = 0; i < iEdges; i++){
+				
+				nodesDualFace.push_back(meshInputDual.nodes[faceDualFace[i]]);
+				
+			}
+			
+			//Vector of contributing weights
+			
+			std::vector<double> vecWeights(iEdges);
+			
+			//Compute generalized barycentric coordinates
+			
+			GeneralizedBarycentricCoordinates(nodeQ, nodesDualFace, vecWeights);	
+		
+			//Add weights to map
+			
+			for (int i = 0; i < iEdges; i++){
 
-						vecTriangleSubAreas[m] = dSubAreaM;
+				int iContributingFaceI = meshInputDual.faces[iFaceFinal][i];
 
-						double dTriangleAreaM = CalculateFaceArea(faceTriangleM, nodesTriangleM);
+				smatMap(ixFirst, iContributingFaceI) = vecWeights[i];
+				
+			}
+			
+		}
+		
+	}
+	
+	else{
```

Comment:
Can you add a comment that this is the else block for mesh with holes. It makes readability better